File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1

        while l <= r:
            mid = (l + r) / 2

            if nums[mid] > nums[target]:
                r = mid - 1
            elif nums[mid] < nums[target]:
                l = mid + 1
            else:
                return mid

        return -1

File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        if (nums.size() == 0)
        {
            return false;
        }

        for (int i = 0; i < nums.size() - 1; i++)
        {
            for (int j = 1; j <= nums.size() - 1; j++)
            {
                if (i != j)
                {
                    if (nums[i] == nums[j])
                    {
                        return true;
                    }
                }
            }
        }
        
        return false;
    }
};
